package.json

```diff
@@ -87,22 +88,38 @@
     "web_modules/",
     "schemas/"
   ],
+  "jest": {
+    "cache": true,
+    "collectCoverageFrom": [
+      "lib/**/*.js",
+      "!**/*.runtime.js"
+    ],
+    "testEnvironment": "node",
+    "testRegex": "test/.*?\\.(benchmark|test)\\.js$",
+    "testPathIgnorePatterns": [
+      "test/binCases",
+      "test/configCases",
+      "test/watchCases",
+      "test/hotCases",
+      "test/js"
+    ]
+  },
   "scripts": {
-    "test": "mocha test/*.test.js --harmony --check-leaks",
-    "travis:test": "npm run cover:min",
+    "pretest": "npm run lint",
+    "test": "npm run unit",
+    "unit": "jest --testPathPattern test\\.js",
+    "benchmark": "jest --testPathPattern benchmark\\.js --verbose",
+    "coverage": "jest --coverage --testPathPattern test\\.js",
+    "travis:test": "npm run coverage",
     "travis:lint": "npm run lint-files && npm run nsp",
     "travis:benchmark": "npm run benchmark",
-    "appveyor:test": "node --max_old_space_size=4096 node_modules\\mocha\\bin\\mocha --harmony test/*.test.js",
+    "appveyor:test": "npm run unit",
     "appveyor:benchmark": "npm run benchmark",
     "build:examples": "cd examples && node buildAll.js",
-    "pretest": "npm run lint-files",
     "lint-files": "npm run lint && npm run beautify-lint",
     "lint": "eslint lib bin hot buildin \"test/**/webpack.config.js\" \"test/binCases/**/test.js\" \"examples/**/webpack.config.js\"",
     "beautify-lint": "beautify-lint \"lib/**/*.js\" \"hot/**/*.js\" \"bin/**/*.js\" \"benchmark/*.js\" \"test/*.js\"",
     "nsp": "nsp check --output summary",
-    "benchmark": "mocha test/*.benchmark.js --harmony -R spec",
-    "cover": "node --harmony ./node_modules/istanbul/lib/cli.js cover -x '**/*.runtime.js' node_modules/mocha/bin/_mocha -- test/*.test.js",
-    "cover:min": "node --harmony ./node_modules/.bin/istanbul cover -x '**/*.runtime.js' --report lcovonly node_modules/mocha/bin/_mocha -- test/*.test.js",
-    "publish-patch": "npm run lint && npm run beautify-lint && mocha && npm version patch && git push && git push --tags && npm publish"
+    "publish-patch": "npm run beautify-lint && npm test && npm version patch && git push && git push --tags && npm publish"
```

Comment:
`npm test` is run in it, and `npm` automatically runs any script target with a "pre" prefix that matches the given target. Therefore, `npm test` will also run `npm run pretest`, which triggers `npm run lint`.